src/ralph/deployment/models.py

```diff
@@ -59,13 +111,43 @@ class Preboot(NamedMixin):
         default='',
     )
 
+    used_counter = models.PositiveIntegerField(default=0, editable=False)
+
     class Meta:
         verbose_name = _('preboot')
         verbose_name_plural = _('preboots')
         ordering = ('name',)
 
+    def increment_used_counter(self):
+        self.used_counter = F('used_counter') + 1
+        self.save()
+
+    def _get_item(self, model_name, item_type):
+        item = None
+        try:
+            queryset_kwargs = {
+                '{}__type'.format(model_name): PrebootItemType.id_from_name(item_type)  # noqa
+            }
+            item = self.items.get(**queryset_kwargs)
+        except PrebootItem.DoesNotExist:
+            pass
+        return item
+
+    def get_file_url(self, file_type):
+        item = self._get_item(model_name='prebootfile', item_type=file_type)
+        if item is not None and item.file:
+            return item.file.url
+
+    def get_configuration(self, configuration_type):
+        item = self._get_item(
+            model_name='prebootconfiguration',
+            item_type=configuration_type
+        )
+        if item is not None:
+            return item.configuration
+
 
-class DeploymentManager(models.Manager):
+class DeploymentManager(JobManager):
     def get_queryset(self):
         from ralph.deployment.deployment import deploy
         # TODO: test it
```

Comment:
add another TODO that this should return only Jobs for DataCenterAsset or VirtualServer